editor/reducer.js

```diff
@@ -414,6 +428,17 @@ export function hoveredBlock( state = null, action ) {
 	return state;
 }
 
+export function blocksMode( state = {}, action ) {
+	if ( action.type === 'TOGGLE_BLOCK_MODE' ) {
+		return {
+			...state,
+			[ action.uid ]: state[ action.uid ] && state[ action.uid ] === 'html' ? 'visual' : 'html',
```

Comment:
Maybe a destructure could help clear this up a bit:

```js
export function blocksMode( state = {}, action ) {
	const { type, uid } = action;
	if ( type === 'TOGGLE_BLOCK_MODE' ) {
		return {
			...state,
			[ uid ]: state[ uid ] && state[ uid ] === 'html' ? 'visual' : 'html',
		};
	}

	return state;
}
```

Or inside the condition if we want to avoid assignment for unhandled actions:


```js
export function blocksMode( state = {}, action ) {
	if ( action.type === 'TOGGLE_BLOCK_MODE' ) {
		const { uid } = action;
		return {
			...state,
			[ uid ]: state[ uid ] && state[ uid ] === 'html' ? 'visual' : 'html',
		};
	}

	return state;
}
```